modules/virt-using-wasp-agent-to-configure-higher-vm-workload-density.adoc

```diff
@@ -0,0 +1,298 @@
+====
+
+.Prerequisites
+
+* The `oc` tool is available.
+* You are logged into the cluster with the cluster-admin role.
+* A memory over-commit ratio is defined. The default is 150%.
+* The node belongs to a worker pool.
+
+.Procedure
+
+[NOTE]
+====
+`wasp-agent` deploys an OCI hook to enable swap usage for containers on the node level. The low-level nature requires the `DaemonSet` object to be privileged.
+====
+
+. Create a privileged service account as follows:
++
+[source,terminal]
+----
+$ oc adm new-project wasp
+$ oc create sa -n wasp wasp
+$ oc adm policy add-cluster-role-to-user cluster-admin -n wasp -z wasp
+$ oc adm policy add-scc-to-user -n wasp privileged -z wasp
+----
+. Deploy `wasp-agent`:
+.. Create a `DaemonSet` object as follows:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: wasp-agent
+  namespace: wasp
+  labels:
+    app: wasp
+    tier: node
+spec:
+  selector:
+    matchLabels:
+      name: wasp
+  template:
+    metadata:
+      annotations:
+        description: >-
+          Configures swap for workloads
+      labels:
+          name: wasp
+    spec:
+      serviceAccountName: wasp
+      hostPID: true
+      hostUsers: true
+      terminationGracePeriodSeconds: 5
+      containers:
+        - name: wasp-agent
+          image: >-
+            registry.redhat.io/container-native-virtualization/wasp-agent-rhel9:latest
+          imagePullPolicy: Always
+          env:
+          - name: "FSROOT"
+            value: "/host"
+          resources:
+            requests:
+              cpu: 100m
+              memory: 50M
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - name: host
+            mountPath: "/host"
+      volumes:
+      - name: host
+        hostPath:
+          path: "/"
+      priorityClassName: system-node-critical
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+      maxSurge: 0
+status: {}
+----
+.. Get the latest URL from the link:https://registry.redhat.io/container-native-virtualization/wasp-agent-rhel9:latest[RH portal] and enter the URL in the created `DaemonSet` object.
+. Configure the `kubelet` service to permit swap:
+.. Create a `KubeletConfiguration` file as follows:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: custom-config
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      pools.operator.machineconfiguration.openshift.io/worker: ''  # MCP
+      #machine.openshift.io/cluster-api-machine-role: worker # machine
+      #node-role.kubernetes.io/worker: '' # node
+  kubeletConfig:
+    failSwapOn: false
+    evictionSoft:
+      memory.available: "1Gi"
+    evictionSoftGracePeriod:
+      memory.available: "10s"
+----
++
+If the cluster is already using an existing `KubeletConfiguration` file, add the following to the spec:
```

Comment:
```suggestion
.. Optional step. If the cluster is already using an existing `KubeletConfiguration` file, add the following to the `spec` section:
```

Reply:
@dfitzmau I am not sure I can agree with your suggestion. This is not an optional step. It is a mandatory, albeit conditioned, step.